analyzers/tests/SonarAnalyzer.UnitTest/TestCases/ParametersCorrectOrder.vb

```diff
@@ -68,4 +70,65 @@
             End Sub
         End Class
     End Class
+
+    ' See https://github.com/SonarSource/sonar-dotnet/issues/3879
+    Public Class NotOnlyNullableParam
+
+        Public Class A
+            Public Property Something As C
+        End Class
+
+        Public Class C
+            Public Property b As Integer
+        End Class
+
+        Public Class B
+            Public Property Value As Integer
+        End Class
+
+        Sub NotNullableParamVoid(ByVal a As Integer, ByVal b As Integer)
+            ' Do nothing
+        End Sub
+
+        Sub NullableParamValueVoid(ByVal a As Integer, ByVal b As Nullable(Of Integer))
+
+            If b.HasValue Then
+                NotNullableParamVoid(b.Value, a) ' Noncompliant
+                NotNullableParamVoid(a, b.Value) ' Compliant
+            End If
+        End Sub
+
+
+        Sub NullableParamCastVoid(ByVal a As Integer, ByVal b As Integer?)
+            If b.HasValue Then
+                NotNullableParamVoid(CInt(b), a) ' Noncompliant
```

Comment:
There should be also cases with `DirectCast` and `CType`

Reply:
`TryCast` accepts a reference type, while `DirectCast` can't convert a nullable int to an int (error BC30311)